plugins/outputs/influxdb/udp.go

```diff
@@ -96,7 +98,11 @@ func (c *udpClient) Write(ctx context.Context, metrics []telegraf.Metric) error
 			continue
 		}
 
-		_, err = c.conn.Write(octets)
+		scanner := bufio.NewScanner(bytes.NewReader(octets))
+		scanner.Split(scanLines)
+		for scanner.Scan() {
+			_, err = c.conn.Write(scanner.Bytes())
+		}
```

Comment:
For 1.10, I think I will experiment with doing a metric splitter before the Serialize step, so we don't need to parse the serialized data.  There might be some duplication of the serialize code but I think it will be worth it, and also string fields can contain unescaped newlines in line protocol so there is a potential flaw with this (#4942).

I think we are going to do a 1.9.5, and this will be a more appropriate change for it.

Reply:
I completely agree doing the change you proposed is a sensible approach. This was just a stopgap to get things working with minimal modifications to the existing code.